tools/rebuild_provision_request.rb

```diff
@@ -0,0 +1,305 @@
+  end
+
+  def dialog_field_values(dialog_tab)
+    dialog_tab[:fields].keys.each do |field|
+      value = @provision_options[field]
+      value = value.first if value.kind_of?(Array)
+      yield(field.to_s, value) if value.present?
+      @provision_options.delete(field)
+    end
+  end
+
+  def process_dialog_tab(tab, dialog_tab)
+    @log += "Processing tab: #{tab}\n" if @verbose
+    @log += "\n" if tab == 'customize' && @verbose
+    tab_name = respond_to?(tab, true) ? tab : :general_tab
+    send(tab_name, tab, dialog_tab)
+  end
+
+  def requester(_tab, dialog_tab)
+    request = @output["requester"]
+    dialog_field_values(dialog_tab) do |field, value|
+      request[field] = value
+    end
+  end
+
+  def purpose(_tab, _dialog_tab)
+    request = @output["tags"]
+    tag_ids = @provision_options.delete(:vm_tags)
+    Classification.where(:id => tag_ids).each do |tag|
+      puts "Found tag: #{tag.name} - Adding parent tag: #{tag.parent.name}" if @verbose
+      request[tag.parent.name] = tag.name
+    end
+  end
+
+  def environment(_tab, _dialog_tab)
+    request = @output["vm_fields"]
+    if @provision_options[:placement_auto].first == true
+      request["placement_auto"] = true
+      return
+    end
+
+    request["placement_auto"] = false
+  end
+
+  def general_tab(_tab, dialog_tab)
+    request = @output["vm_fields"]
+    dialog_field_values(dialog_tab) do |field, value|
+      request[field] = value
+    end
+  end
+
+  def customize(tab, dialog_tab)
+    @provision_options.delete(:customization_template_script)
+    general_tab(tab, dialog_tab)
+  end
+
+  def schedule(tab, dialog_tab)
+    if @provision_options[:schedule_type].first == "immediately"
+      @provision_options.delete(:schedule_type)
+      @provision_options.delete(:schedule_time)
+    else
+      @provision_options[:schedule_time] = @provision_options[:schedule_time].to_s
+    end
+    general_tab(tab, dialog_tab)
+  end
+end
+
+class Dialog
+  attr_reader :provision_options
+
+  def initialize(provision_options, verbose)
+    @verbose = verbose
+    @provision_options = provision_options
+    load_dialog_content
+  end
+
+  def load_dialog_content
+    dialog_name = @provision_options.delete(:miq_request_dialog_name)
+    @dialog = MiqDialog.find_by(:name => dialog_name)
+  end
+
+  def content
+    @dialog.content
+  end
+end
+
+class AutomateHash
+  attr_reader :request_id
+
+  URL_PATH = "/api/provision_requests".freeze
+
+  def initialize(request_id, opts)
+    @verbose = opts[:verbose]
+    @options = opts
+    @request_id = request_id
+  end
+
+  def last_requests
+    print "\nSearching for the last #{@options[:last_requests]} requests run against #{build_url}\n\n" if @verbose
+    list = MiqProvisionRequest.limit(@options[:last_requests]).order(:created_on => 'DESC').collect(&:id)
+    if @verbose
+      print "Found these #{list.count} request ids: #{list.join(',')}\n\n"
+    else
+      puts list.join(',')
```

Comment:
Would be nice to output the list with `ID - Description` as the default.  Then support a limited output with just the IDs.

```ruby
list = MiqProvisionRequest.limit(@options[:last_requests]).order(:created_on => 'DESC').pluck(:id, :description)
list.each {|l| puts l.join(" - ")}
```